### PR TITLE
Fixes #27

### LINKED ISSUE #27: Test pretty printing
While working on #24 it turned out that the current test suite performed poorly.
Proposal:

1. Drop a bash snippet somewhere in the tests/ directory
2. Discover those snippets
3. Parse them
4. Pretty print them
5. Compare the output to a golden file using [tasty-golden](https://hackage.haskell.org/package/tasty-golden)
6. Run bash on the original file and the pretty printed one and compare the output

This is of course only for those tests involving pretty printing; we can keep the existing unit tests as they are.
Additionally we could fuzzy test the implementation with QuickCheck.

### PR DESCRIPTION
**Pull request by mmhat**
Please do not merge until we settle for an appropriate pretty printer output (#24).

### CODE REVIEW

**Comment by mmhat** [comment]
@knrafto [tests/pretty/heredoc-nested.sh](https://github.com/knrafto/language-bash/pull/30/commits/ed3bf8925a83b8675d08065d63aaa368b24669eb#diff-e01140c6a9fb08ef3c7c2f90234f22f6) cannot be parsed but 'bash -n' says it's ok. Can you have a look?

**Comment by knrafto** [comment]
This looks fantastic, thank you so much!
Thanks for finding the bug too, I pushed a fix to master.

File: tests/Tests.hs [hunk 1]
```diff
@@ -58,6 +62,23 @@ prop_expandsLikeBash = monadicIO $ forAllM braceExpr $ \str -> do
 properties :: TestTree
 properties = testGroup "Properties" [testProperty "brace expansion" prop_expandsLikeBash]
 
+discoverPretty :: FilePath -> IO TestTree
```
Comment (knrafto) [inline tests/Tests.hs]: nit: maybe `discoverPrettyTests`

Comment (mmhat) [reply tests/Tests.hs]: Done

**Comment by mmhat** [comment]
Ok, this one is ready to be merged.

**Comment by knrafto** [comment]
Sorry for the flaky test, but this one is a build error. Maybe use `filter (\fn -> takeExtension fn == ".sh")` instead

**Comment by mmhat** [comment]
Ok, done.

**Pull request merged**